### PR TITLE
input: allow holding down forward/back to move through saves

### LINKED ISSUE #171: Feature request: Allow to hold up/down key in savestate list
Currently have to step through them with key presses. Would be better UX if you could hold up or down. TR2Main added this sometime last year I think.

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #171.

### CODE REVIEW

**Review by rr- — CHANGES_REQUESTED** [review]
I'm glad that you went with the tested TR2Main approach, but it needs a bit of polishing.

- I don't understand why these are negative numbers? Why not counting to `15` from `0`?
- We don't put `static` variables inside functions, let's make it a module level variable to highlight its global scope.
- Similarly we don't name variables with camelCase. This should be `snake_case` for normal variables and `m_PascalCase` for module-level globals.

**Review by rr- — COMMENTED** [review]
I understand that we need initial buildup, what I fail to understand is why we're starting at -12 and waiting till it passes 3 instead of starting at 0 and waiting till it passes 15. Also this should be a constant or a module-level `#define`.

**Comment by walkawayy** [comment]
Hmm ya that is odd. I tried to be consistent with how TR2Main did it. I was thinking maybe the negative number was to indicate that when the value is below 0, there is that initial delay. But that would use values of -15 and 0.

**Comment by rr-** [comment]
Yes, it will need to be scaled.

File: src/game/input.c [hunk 1]
```diff
@@ -2,16 +2,37 @@
 
 #include "specific/s_input.h"
 
+#define DELAY_FRAMES 12
+#define HOLD_FRAMES 3
```
Comment (rr-) [inline src/game/input.c:6]: OK now it's readable, we wait 12 frames till letting the user cycle fast, and then we only do it once every 3 frames.

File: src/game/input.c [hunk 2]
```diff
@@ -2,16 +2,37 @@
...
+
 INPUT_STATE g_Input = { 0 };
 INPUT_STATE g_InputDB = { 0 };
 INPUT_STATE g_OldInputDB = { 0 };
+static int32_t m_HoldBack = 0;
```
Comment (rr-) [inline src/game/input.c:11]: Let's put a blank above since the vars before are truly global, and these are only modular

**Review by rr- — COMMENTED** [review]
Thanks for the refactor

File: src/game/input.c (hunk 1 above)
Comment (walkawayy) [reply src/game/input.c:6]: Ya. Then we can just scale this for 60 FPS in the future.

**Review by rr- — APPROVED** [review]

**Comment by Richard-L** [comment]
@walkawayy any chance you could look into this again? 🙏 It seems to have broken in v2.15

**Pull request merged**